gobblin-data-management/src/main/java/org/apache/gobblin/data/management/source/LoopingDatasetFinderSource.java

```diff
@@ -105,11 +120,16 @@ public WorkUnitStream getWorkunitStream(SourceState state) {
 
       IterableDatasetFinder datasetsFinder = createDatasetsFinder(state);
 
-      Stream<Dataset> datasetStream = datasetsFinder.getDatasetsStream(Spliterator.SORTED, this.lexicographicalComparator);
+      Stream<Dataset> datasetStream =
+          datasetsFinder.getDatasetsStream(Spliterator.SORTED, this.lexicographicalComparator);
       datasetStream = sortStreamLexicographically(datasetStream);
 
-      return new BasicWorkUnitStream.Builder(new DeepIterator(datasetStream.iterator(), previousDatasetUrnWatermark,
-          previousPartitionUrnWatermark, maxWorkUnits)).setFiniteStream(true).build();
+      Config config = ConfigUtils.propertiesToConfig(state.getProperties());
+      String storeName = JobState.getJobNameFromState(state);
+
+      return new BasicWorkUnitStream.Builder(
+          new DeepIterator(datasetStream.iterator(), previousDatasetUrnWatermark, previousPartitionUrnWatermark,
```

Comment:
`previousDatasetUrnWatermark` is still determined from the largest ordinal in the previous run? Doesn't this need to be updated?

Reply:
Just wanted to leave a quick response here. When datasetstatestore is enabled, the ordinal is effectively useless, since there will be only one workunit passed to the source (i.e. the globalwatermark dataset) in the getPreviourWorkUnitStates() call, which will end up as the maxWorkUnit. The intent was to make minimal changes to allow enabling/disabling dataset state store.